src/lib/rng/system_rng/system_rng.cpp

```diff
@@ -48,9 +49,24 @@ class System_RNG_Impl final : public RandomNumberGenerator
 
       void randomize(uint8_t buf[], size_t len) override
          {
-         bool success = m_rtlgenrandom(buf, ULONG(len)) == TRUE;
-         if(!success)
-            throw System_Error("RtlGenRandom failed");
+         const uint32_t limit = std::numeric_limits<ULONG>::max();
```

Comment:
If this was a `size_t` you can avoid having to specify the type for `std::min(byteLeft, limit)`